graphql_compiler/compiler/expressions.py

```diff
@@ -233,12 +233,21 @@ def __ne__(self, other):
 class LocalField(Expression):
     """A field at the current position in the query."""
 
-    __slots__ = ('field_name',)
+    __slots__ = ('field_name', 'field_type')
 
-    def __init__(self, field_name):
-        """Construct a new LocalField object that references a field at the current position."""
-        super(LocalField, self).__init__(field_name)
+    def __init__(self, field_name, field_type):
+        """Construct a new LocalField object that references a field at the current position.
+
+        Args:
+            field_name: string, the name of the local field being referenced
+            field_type: GraphQL type object describing the type of the referenced field. For some
```

Comment:
Two nits. GraphQLType not GraphQL Type.  If the field can be None, please start with the word optional.

Reply:
I would disagree on the "optional" part, and this is something that will get better once we drop support for Python 2.x and enforce type hints everywhere. The type of the value is a "maybe" type (`Optional[GraphQLType]`), but the argument is not optional -- it is positional and required, and it's allowed in some circumstances to supply the value `None`.

I would prefer to avoid describing it as "optional" since it's not an optional argument, but a required argument with an optional type.